src/main.rs

```diff
@@ -63,6 +74,216 @@ async fn init_maker(
     }
 }
 
+// TODO: consider introducing components for this (also balance updates below),
+//  Rough idea: Extend MidMarketRate (or introduce new struct) to give us an async next function, hide the fact that it is using channels behind the next function of the component.
```

Comment:
I don't think `next` is the appropriate terminology, it is not an iterator.

Reply:
I just removed the `TODO` for now, no need to deal with this before the project finishes.